### PR TITLE
Feature/max avg apy staking

### PR DESCRIPTION
**Pull request by FrogTravel**
Change the UI for APY and APR on staking screen. Maximum APY in the BottomSheet is mock

### CODE REVIEW

File: common/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1,5 +1,13 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <resources>
+    <string name="staking_reward_estimation_bottom_sheet_title">Earnings in the network</string>
+
+    <string name="staking_only_apr">APR</string>
+    <string name="staking_only_apy">APY</string>
```
Comment (valentunn) [inline common/src/main/res/values/strings.xml:6]: We already have

```
   <string name="staking_apy">%s APY</string>
    <string name="staking_apr">%s APR</string>
```

File: feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/setup/view/RewardDestinationView.kt [hunk 2]
```diff
@@ -59,6 +57,10 @@ class RewardDestinationView @JvmOverloads constructor(
         payoutTargetAmountGain.text = gain
     }
 
+    fun setFiatAmount(amount: String?) {
+        if (amount == null) payoutTargetAmountFiat.makeGone() else payoutTargetAmountFiat.text = amount
```
Comment (valentunn) [inline feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/setup/view/RewardDestinationView.kt:61]:
```suggestion
       payoutTargetAmountFiat.setTextOrHide(amount)
```

File: feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/staking/main/StakingViewState.kt [hunk 3]
```diff
@@ -259,6 +262,11 @@ class WelcomeViewState(
         ReturnsModel(monthlyEstimation, yearlyEstimation)
     }.asLiveData(scope)
 
+    fun infoActionClicked() {
+        val reward = returns.value
```
Comment (valentunn) [inline feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/staking/main/StakingViewState.kt:266]: You can do `val reward = returns.value ?: return` to get rid of nested block on next line

File: feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/staking/main/model/RewardEstimation.kt [hunk 4]
```diff
@@ -3,5 +3,5 @@ package jp.co.soramitsu.feature_staking_impl.presentation.staking.main.model
 class RewardEstimation(
     val amount: String,
     val fiatAmount: String?,
-    val gain: String,
+    val gain: String, // TODO
```
Comment (valentunn) [inline feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/staking/main/model/RewardEstimation.kt:6]: What does  `TODO` mean here?

File: feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/staking/main/view/StakingInfoView.kt [hunk 5]
```diff
@@ -71,6 +79,8 @@ class StakingInfoView @JvmOverloads constructor(
     }
 
     fun setExtraBlockAdditionalText(text: String) {
+        stakingInfoExtraTitleDetailView.visibility = View.VISIBLE
```
Comment (valentunn) [inline feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/staking/main/view/StakingInfoView.kt:82]: stakingInfoExtraTitleDetailView.makeVisible()

File: feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/staking/main/StakingViewState.kt [hunk 6]
```diff
@@ -259,6 +262,11 @@ class WelcomeViewState(
...
+        reward?.let { _showRewardEstimationEvent.value = Event(StakingRewardEstimationBottomSheet.Payload(reward.monthly, reward.yearly)) }
```
Comment (valentunn) [inline feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/staking/main/StakingViewState.kt:267]: To calculate max apy, you should add `getMaxApy(): Double` method to  reward calculator, which should just take maximum value from `apyByValidator` mapping

**Review by valentunn — CHANGES_REQUESTED** [review]

File: common/src/main/res/values/strings.xml (hunk 1 above)
Comment (FrogTravel) [reply common/src/main/res/values/strings.xml:6]: I don't need a template here, just APY and APR, 'cause I use two TextView to display "1% APY"

Comment (valentunn) [reply common/src/main/res/values/strings.xml:6]: Yeah, I meant that we could reuse existing strings to not to have a lot of dublicates

File: feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/staking/main/StakingViewState.kt (hunk 6 above)
Comment (FrogTravel) [reply feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/staking/main/StakingViewState.kt]: Done

File: feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/staking/main/view/StakingInfoView.kt (hunk 5 above)
Comment (FrogTravel) [reply feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/staking/main/view/StakingInfoView.kt]: Done

File: feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/staking/main/model/RewardEstimation.kt (hunk 4 above)
Comment (FrogTravel) [reply feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/staking/main/model/RewardEstimation.kt]: Done

File: feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/staking/main/StakingViewState.kt (hunk 3 above)
Comment (FrogTravel) [reply feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/staking/main/StakingViewState.kt]: Done

File: feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/setup/view/RewardDestinationView.kt (hunk 2 above)
Comment (FrogTravel) [reply feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/presentation/setup/view/RewardDestinationView.kt]: Done

**Review by valentunn — APPROVED** [review]

**Pull request merged**